src/core/ledger/grainAllocation.js

```diff
@@ -214,12 +222,13 @@ function receipts(
  */
 function immediateReceipts(
   budget: G.Grain,
-  identities: ProcessedIdentities
+  identities: ProcessedIdentities,
+  mapping?: Function = (id) => id
 ): $ReadOnlyArray<GrainReceipt> {
-  const amounts = G.splitBudget(
-    budget,
-    identities.map((i) => i.mostRecentCred)
-  );
+  const credPerIdentity = identities.map((i) => i.mostRecentCred);
+  // if (!mapping) mapping = (id) => id
+  const modifiedCred = credPerIdentity.map(mapping);
```

Comment:
I think the mapping function should accept the entire array of cred scores instead of each individual cred score. That way it can have context for what the entire range of values are and allow for more advanced mapping strategies (e.g. bell curve)